### PR TITLE
Home work1 - completed

### PR DESCRIPTION
**Pull request by daarklight**
This is 1st trial.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,16 +1,74 @@
...
+      <artifactId>calculator</artifactId>
+      <scope>system</scope>
+      <version>1.0</version>
+      <systemPath>${project.basedir}/src/main/resources/calculator-1.0.jar</systemPath>
+    </dependency>
+
+    <dependency>
+      <groupId>org.testng</groupId>
+      <artifactId>testng</artifactId>
+      <version>7.4.0</version>
```
Comment (shoggy) [inline pom.xml]: `properties` section is better place for versions

File: pom.xml [hunk 2]
```diff
@@ -1,16 +1,74 @@
...
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>4.13</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (shoggy) [inline pom.xml]: What is this dependency required for?

File: pom.xml [hunk 3]
```diff
@@ -1,16 +1,74 @@
...
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <version>3.0.0-M5</version>
+        <configuration>
+          <suiteXmlFiles>
+            <suiteXmlFile>${maven-surefire-plugin.suiteXmlFile}</suiteXmlFile>
+          </suiteXmlFiles>
+        </configuration>
+      </plugin>
+    </plugins>
```
Comment (shoggy) [inline pom.xml:84]: `maven-checkstyle-plugin` definition is missed

File: pom.xml [hunk 4]
```diff
@@ -1,16 +1,74 @@
...
+    </profile>
+
+    <profile>
+      <id>allTests</id>
+      <properties>
+        <suiteXMLfile>${project.basedir}/src/test/java/ru/training/at/hw1/resources/AllTests.xml</suiteXMLfile>
+      </properties>
+    </profile>
+
+  </profiles>
```
Comment (shoggy) [inline pom.xml]: Three profiles are required

File: src/test/java/ru/training/at/hw1/mainTestClasses/AdditionTest.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import ru.training.at.hw1.DataProviders;
+
+public class AdditionTest {
+
+    @Test(dataProviderClass = DataProviders.class, dataProvider = "addition data - long",
+          groups = "additionAndSubtraction")
+    public void additionTestingLong(long firstNum, long secondNum, long expectedResult) {
+        Calculator calculator = new Calculator();
```
Comment (shoggy) [inline src/test/java/ru/training/at/hw1/mainTestClasses/AdditionTest.java]: Looks like SystemUnderTest-initialization, please move it to the appropriate place. Pay attention that all Test Classes have this point in common

File: src/test/java/ru/training/at/hw1/mainTestClasses/DivisionTest.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        Assert.assertEquals(actualResult, expectedResult);
+        System.out.println(firstNum + " / " + secondNum + " = " + expectedResult);
+    }
+
+    @Test(dataProviderClass = DataProviders.class, dataProvider = "division data - double",
+          groups = "divisionAndMultiplication")
+    public void divisionTestingDouble(double firstNum, double secondNum, double expectedResult) {
+        Calculator calculator = new Calculator();
+        double actualResult = calculator.div(firstNum, secondNum);
+        Assert.assertEquals(actualResult, expectedResult);
```
Comment (shoggy) [inline src/test/java/ru/training/at/hw1/mainTestClasses/DivisionTest.java]: Double comparison is not strict and requires one additional parameter

**Review by shoggy — CHANGES_REQUESTED** [review]

**Comment by daarklight** [comment]
This is 2nd trial. All comments were taken into account.

**Review by shoggy — APPROVED** [review]
Good work
In the next tasks please avoid `System.out` and magic numbers

**Pull request merged**